### ISSUE TITLE
News sitemap timestamp doesn't match with site's local timezone. 

### ISSUE THREAD

**Issue by StefMattana** [Automattic/jetpack#36217]
### Impacted plugin

Jetpack

### Quick summary

The news sitemap created by Jetpack reports a publishing time that doesn't match the site's. 
The news sitemap is set to ICT while it should be on IST.

The issue has been reported [in this forum thread](https://wordpress.org/support/topic/publication-date-time-zone/#post-17477860). 

### Steps to reproduce

- go to https://fmsikar.in/news-sitemap.xml
- check one entry and the related timestamp; I checked [this post](https://fmsikar.in/entertainment/anant-ambani-radhika-merchant-pre-wedding-total-expenses-know-why-the-people-convinced-of-the-ambani-family), whose publication time in the sitemap is `2024-03-06T11:03:29Z`
- go to the post and check the publication time; in my case, the post states `Last updated: 06, March, 2024 4:33 PM`

### A clear and concise description of what you expected to happen.

I expected the timestamp to be the same. 

### What actually happened

There is a discrepancy of ~5 hours. 
A couple of notes: 

- The user had the news sitemap issued by RankMath, which had the very same issue. [Upon contacting its support team](https://support.rankmath.com/ticket/news-sitemap-45/?view=all&unm=726003), they opted to enable Jetpack's news sitemap again. 
- the fact that both news sitemaps reported the same time discrepancy made me think it could be a site-related issue. 
- The site has the timezone settings on the IST time. Also the server settings have been checked. 
- I asked the user whether the posts had been edited after being published to rule out any discrepancies caused by that. The time discrepancy doesn't look perfect (ie an exact 5 hours of difference, but it's always a bit more or a bit less - for example, sitemaps timestamping `11:03:29` and the post's metadata reporting `4:33`).
- I also asked to do some tests to avoid plugin conflicts. 

### Impact

One

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple, Self-hosted

### Logs or notes

DB green: https://jptools.wordpress.com/debug/?url=%20http%3A%2F%2Ffmsikar.in
Forum thread: https://wordpress.org/support/topic/publication-date-time-zone/#post-17477860

**Comment by liviopv** [comment]
📌 REPRODUCTION RESULTS
- Tested on Atomic – Uncertain

📌 FINDINGS/SCREENSHOTS/VIDEO
@StefMattana I published and updated a post on my site and the sitemap showed the correct timestamp
I have now switched the timezone of the site to the same as the user in this thread.

Sitemap: https://atlivio.blog/news-sitemap.xml
Post: https://atlivio.blog/2024/03/06/jetpack-sitemap-test/

I'm leaving this for now to give the sitemap time to update, but everything points to it being a site-specific issue.

Looking at the site's feed, the `pubDate` in the feed is the same as the one in the sitemap `<pubDate>Wed, 06 Mar 2024 11:03:29 +0000</pubDate>`, which indicates it should be just that the sitemap is showing publication date and the frontend is showing Last Edited time.

📌 ACTIONS
- Requested author feedback

**Comment by anomiex** [comment]
I'm finding myself a little confused as to what's supposed to be going on here. From the reproduction instructions,

> 2024-03-06T11:03:29Z

> 06, March, 2024 4:33 PM

Assuming the second is India Standard Time (IST), those are the same (with the second dropping the seconds). The "Z" suffix on the first indicates it's in UTC, and 11:03 UTC is indeed 4:33 PM IST.

I'm not seeing where ICT (Indochina Time) comes into it at all. ICT for that time would be 6:03 PM (18:03 in 24-hour format).

> The time discrepancy doesn't look perfect (ie an exact 5 hours of difference

Note IST is UTC+5:30.

In machine-readable data it's fairly common to output timestamps in UTC, relying on the client consuming the data to convert it to the appropriate timezone for display to humans. The sitemap code in Jetpack does this, and I see no option for outputting the data using a timezone other than UTC.

**Comment by StefMattana** [comment]
Hey @anomiex, 

> The "Z" suffix on the first indicates it's in UTC, and 11:03 UTC is indeed 4:33 PM IST.

TIL 🤯 

> I'm not seeing where ICT (Indochina Time) comes into it at all. ICT for that time would be 6:03 PM (18:03 in 24-hour format).

This is what I saw in the previous RankMath and Jetpack sitemaps before logging the issue here. I nudged the user to check their timezone settings, they said it was set on IST while I checked that the timezone had ~ hours discrepancy with UTC 🤷 

It sounds like we solved the mystery, and then there's nothing actionable from our end. Thanks all!
